core/modules/system/system.install

```diff
@@ -14,7 +14,129 @@ function system_requirements($phase) {
   // Ensure translations don't break during installation.
   $t = get_t();
 
-  // Report Backdrop version
+  // System metrics.
+  if ($phase == 'runtime') {
+    $metrics = array();
+
+    $free_output = shell_exec('free');
+    $free_output = (string) trim($free_output);
+    $free_lines = explode("\n", $free_output);
+
+    $result = array();
+    foreach ($free_lines as $line => $output) {
+      // Convert to lowercase.
+      $output = strtolower($output);
+      // Remove multiple spaces.
+      $output = preg_replace('/[\s]+/mu', ' ', $output);
+      // Convert any special characters to underscore.
+      $output = preg_replace('/[^a-zA-Z0-9_\s]/', '_', $output);
+
+      $result[$line] = explode(" ", $output);
+    }
+
+    // Grab the 1st line of the output (which is the heading of the various
+    // columns), then remove that from the rest of the output.
+    $header = $result[0];
+    unset($result[0]);
+    // Grab each RAM metric (per column).
+    $mem_metrics = array();
+    foreach ($result as $result_line => $data) {
+      // The first column contains the type of memory ("mem" or "swap").
+      $prefix = $data[0];
+      array_splice($data, 0, 1);
+      foreach ($data as $column => $metric) {
+        $mem_metrics[$prefix . $header[$column]] = $metric;
+      }
+    }
+
+    $mem_usage = round(($mem_metrics['mem_total'] - $mem_metrics['mem_available']) / $mem_metrics['mem_total'] * 100);
+
+    // Disk metrics. Should work the same across operating systems.
+    $disk_total = disk_total_space('.');
+    $disk_free = disk_free_space('.');
+    $disk_used = $disk_total - $disk_free;
+    $disk_usage = round($disk_used / $disk_total * 100);
+
+    // Sizes of the various files directories.
+    $directories = array(
+      'temp' => file_directory_temp(),
+      'public' => backdrop_realpath('public://'),
+      'private' => backdrop_realpath('private://'),
+    );
+    $directory_sizes = array();
+    $total_dir_size = 0;
+    foreach ($directories as $directory => $path) {
+      if (!empty($path)) {
+        $dir_size = _system_get_dir_size($path);
+        $directory_sizes[$directory] = $t('@dir: @size', array('@dir' => $directory, '@size' => format_size($dir_size)));
+        $total_dir_size += $dir_size;
+      }
+    }
+    $dir_size_breakdown = theme('item_list', array('items' => $directory_sizes));
+
+    // Database metrics.
+    $rows = db_query('SHOW TABLE STATUS')->fetchAll();
```

Comment:
There might be a smarter way to do this.

Reply:
This was based on some quick research on the internet. Happy to be provided with other ways/ideas.

Reply:
A quick experiment:

```
global $databases;
$db_name = $databases['default']['default']['database'];

$size = db_query('SELECT ROUND(SUM(data_length + index_length) / 1024 / 1024, 2) as size_mb FROM information_schema.tables WHERE table_schema = :dbname GROUP BY table_schema', array(':dbname' => $db_name))->fetchField();

debug($size);
```